### PR TITLE
TimeSeries: Allow multiple axes for the same unit

### LINKED ISSUE #38810: GraphNG: Allow axis placement control that's independent from unit config
**What happened**:
Override setting not beeing honerd

**What you expected to happen**:
Field/Query to be moved to the right side

**How to reproduce it (as minimally and precisely as possible)**:
Add override in the panel-settings.

**Anything else we need to know?**:
JSON of panel:
```json
{
  "id": 4,
  "gridPos": {
    "h": 8,
    "w": 16,
    "x": 0,
    "y": 0
  },
  "type": "timeseries",
  "title": "InfluxDB",
  "datasource": "TF influx_1-8-5_prd_primary_v1",
  "fieldConfig": {
    "defaults": {
      "custom": {
        "drawStyle": "line",
        "lineInterpolation": "linear",
        "barAlignment": 0,
        "lineWidth": 2,
        "fillOpacity": 0,
        "gradientMode": "none",
        "spanNulls": false,
        "showPoints": "auto",
        "pointSize": 1,
        "stacking": {
          "mode": "none",
          "group": "A"
        },
        "axisPlacement": "left",
        "axisLabel": "",
        "scaleDistribution": {
          "type": "linear"
        },
        "hideFrom": {
          "tooltip": false,
          "viz": false,
          "legend": false
        },
        "thresholdsStyle": {
          "mode": "off"
        }
      },
      "color": {
        "mode": "palette-classic"
      },
      "thresholds": {
        "mode": "absolute",
        "steps": [
          {
            "color": "green",
            "value": null
          },
          {
            "color": "red",
            "value": 80
          }
        ]
      },
      "mappings": []
    },
    "overrides": [
      {
        "matcher": {
          "id": "byName",
          "options": "Series"
        },
        "properties": [
          {
            "id": "custom.axisPlacement",
            "value": "right"
          }
        ]
      }
    ]
  },
  "options": {
    "tooltip": {
      "mode": "single"
    },
    "legend": {
      "displayMode": "table",
      "placement": "right",
      "calcs": [
        "last"
      ]
    }
  },
  "targets": [
    {
      "alias": "Measurements",
      "groupBy": [
        {
          "params": [
            "1m"
          ],
          "type": "time"
        },
        {
          "params": [
            "null"
          ],
          "type": "fill"
        }
      ],
      "measurement": "influxdb_database",
      "orderByTime": "ASC",
      "policy": "default",
      "queryType": "randomWalk",
      "refId": "A",
      "resultFormat": "time_series",
      "select": [
        [
          {
            "params": [
              "numMeasurements"
            ],
            "type": "field"
          },
          {
            "params": [],
            "type": "mean"
          }
        ]
      ],
      "tags": []
    },
    {
      "alias": "Series",
      "groupBy": [
        {
          "params": [
            "1m"
          ],
          "type": "time"
        },
        {
          "params": [
            "null"
          ],
          "type": "fill"
        }
      ],
      "hide": false,
      "measurement": "influxdb_database",
      "orderByTime": "ASC",
      "policy": "default",
      "queryType": "randomWalk",
      "refId": "B",
      "resultFormat": "time_series",
      "select": [
        [
          {
            "params": [
              "numSeries"
            ],
            "type": "field"
          },
          {
            "params": [],
            "type": "mean"
          }
        ]
      ],
      "tags": []
    },
    {
      "alias": "Shards",
      "groupBy": [
        {
          "params": [
            "1m"
          ],
          "type": "time"
        },
        {
          "params": [
            "null"
          ],
          "type": "fill"
        }
      ],
      "hide": false,
      "measurement": "influxdb",
      "orderByTime": "ASC",
      "policy": "default",
      "queryType": "randomWalk",
      "refId": "C",
      "resultFormat": "time_series",
      "select": [
        [
          {
            "params": [
              "n_shards"
            ],
            "type": "field"
          },
          {
            "params": [],
            "type": "mean"
          }
        ]
      ],
      "tags": []
    }
  ]
}

```

**Environment**:
- Grafana version: Grafana v8.1.2 (103f8fa094)
- Data source type & version: InfluxDB 1.8.5
- OS Grafana is installed on: Ubuntu, Docker
- User OS & Browser: Win10, Windows Edge
- Grafana plugins: -
- Others: -

### PR DESCRIPTION
**Pull request by dprokop**
This PR changes the way we create keys for uPlot scales. It constructs the key based on the following config properties:
- Unit
- Axis placement
- Min/max
- Scale distribution

Thanks to that we can have multiple axes for the same unit with different ranges, distributions, and placement config. 

Brings back what was possible with the old graph panel.

Closes https://github.com/grafana/grafana/issues/38810
Closes https://github.com/grafana/grafana/discussions/38242

Related issues:
https://github.com/grafana/grafana/issues/41282
https://github.com/grafana/grafana/issues/40668
https://github.com/grafana/grafana/issues/39269
https://github.com/grafana/grafana/issues/40907

### CODE REVIEW

**Comment by torkelo** [comment]
nice!

**Comment by dprokop** [comment]
@leeoniya do you think we should also take soft min/max into consideration?

**Comment by leeoniya** [comment]
nice.

i had some concerns around the cursor sync code that i thought i remembered as needing to match up y scale keys, but it looks like not: https://github.com/grafana/grafana/pull/33433/files#diff-57e07daf7609a601c0e85dc9c2122c20dc0f81ce10240849066aecec310a4f0eR248-R249. i swear i remember it working this way at some point :thinking:. and recall very specifically adding uPlot APIs for this because @ryan wanted to do y sync by matched scales & value and not by position. `null` for the y key means it's by position currently, so this PR does not affect syncing behavior.

```js
      scales: [xScaleKey, null as any],
      match: [() => true, () => true],
```

> @leeoniya do you think we should also take soft min/max into consideration?

probably yes. also maybe axis label?

not sure if there are perf implications for having such long strings for scale keys. will make debugging interesting in either case :rofl:

**Comment by dprokop** [comment]
> not sure if there are perf implications for having such long strings for scale keys. will make debugging interesting in either case

haha, thought the same about the length:P I had an idea those keys could be hashed but this would be unnecessary perf overhead, and think about debugging then!

> @leeoniya do you think we should also take soft min/max into consideration?
> probably yes. also maybe axis label?

makes sense, will add these too.

**Comment by leeoniya** [comment]
> I had an idea those keys could be hashed but this would be unnecessary perf overhead

i think this would only need to happen on (re)config. so the perf is prolly okay.

> and think about debugging then!

yeah, that's my main concern, though hashing everything but the unit/side part of it might be sufficient. i guess let's see how just gluing works.

File: packages/grafana-ui/src/components/TimeSeries/utils.ts [hunk 1]
```diff
@@ -131,7 +138,9 @@ export const preparePlotConfigBuilder: UPlotConfigPrepFn<{ sync: DashboardCursor
     field.state!.seriesIndex = seriesIndex++;
 
     const fmt = field.display ?? defaultFormatter;
-    const scaleKey = config.unit || FIXED_UNIT;
+
+    const scaleKey = buildScaleKey(config);
+    console.log('scaleKey', scaleKey);
```
Comment (leeoniya) [inline packages/grafana-ui/src/components/TimeSeries/utils.ts:143]: :eyes:

Comment (dprokop) [reply packages/grafana-ui/src/components/TimeSeries/utils.ts:143]:
```suggestion
```

**Comment by yellowpattern** [comment]
The need for this as a "feature request" means there has been a regression from earlier versions of grafana.

**Comment by oscarkilhed** [comment]
> The need for this as a "feature request" means there has been a regression from earlier versions of grafana.

This is a feature request for the new time series panel in order to get it closer to feature parity with the old graph panel. If you need this functionality in an older version of Grafana you can still use the old graph panel.

**Pull request merged**